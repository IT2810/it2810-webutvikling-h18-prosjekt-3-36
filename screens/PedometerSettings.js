import React from "react";
import { connect } from "react-redux";

import { StyleSheet, Text, View, Alert, Button, TextInput, Platform } from "react-native";
import HeaderButton from "../components/HeaderButton";
import { setGoal } from "../store/actions/Settings-actions";
import { Cell, Section, TableView } from "react-native-tableview-simple";
import Colors from "../constants/Colors";


class PedometerSettings extends React.Component {
  static navigationOptions = ({ state, navigation }) => {
    return {
      headerTitle: "Pedometer goal",
      headerLeft: <HeaderButton onPress={() => navigation.goBack()} title="Cancel" />,
      headerRight: <HeaderButton onPress={() => navigation.state.params.save()} title="Save" />
    };
  };

  constructor(props) {
    super(props);
    this.state = {
      helpAlert: true,
      goal: this.props.goal
    };
  }

  
  componentDidMount() {
    this.props.navigation.setParams({ save: this.save });
  }

  save = () => {
    const { navigation, dispatch } = this.props;
    dispatch(setGoal(this.state.goal));
    navigation.goBack();
  };

  render() { 
    return (
      <View style={{ flex: 1, backgroundColor: "#EFEFF4" }}>
        <TableView>
          <Section>
            <Cell
              cellContentView={
                <TextInput underlineColorAndroid="rgba(0,0,0,0)" style={{ fontSize: 16, flex: 1 }}  keyboardType='numeric' maxLength={10} autoFocus={true} onChangeText={text => this.setState({ goal: text })} value={this.state.goal} placeholder="Goal" />
              }
            />
          </Section>
        </TableView>
      </View>
    );
  }
}

export default connect(state => ({ goal: state.settings.goal }))(PedometerSettings);
