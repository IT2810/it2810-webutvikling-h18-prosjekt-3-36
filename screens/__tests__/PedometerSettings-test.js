import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import PedometerSettings from "../PedometerSettings";
import appState from '../../store/reducers/Reducer'
import { createStore } from 'redux';

const store = createStore(appState)

it("PedometerSettings Matches snapshot.", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<PedometerSettings store={store} navigation={{ getParam: () => { }, setParams: () => { } }} isModalVisible={false} />);

    const renderer2 = new ShallowRenderer();
    renderer2.render(renderer.getRenderOutput());
    const result = renderer2.getRenderOutput();
    expect(result).toMatchSnapshot();
});