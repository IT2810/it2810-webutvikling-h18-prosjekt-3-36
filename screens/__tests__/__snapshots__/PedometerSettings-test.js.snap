// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PedometerSettings Matches snapshot. 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#EFEFF4",
      "flex": 1,
    }
  }
>
  <TableView>
    <Section
      allowFontScaling={true}
      footer={null}
      footerComponent={null}
      footerTextColor="#6d6d72"
      header={null}
      headerComponent={null}
      headerTextColor="#6D6D72"
      hideSeparator={false}
      sectionPaddingBottom={15}
      sectionPaddingTop={15}
      sectionTintColor="#EFEFF4"
      separatorInsetLeft={15}
      separatorInsetRight={0}
      separatorTintColor="#C8C7CC"
    >
      <Cell
        accessory={false}
        accessoryColor="#007AFF"
        accessoryColorDisclosureIndicator="#C7C7CC"
        allowFontScaling={true}
        backgroundColor="#FFF"
        cellAccessoryView={null}
        cellContentView={
          <TextInput
            allowFontScaling={true}
            autoFocus={true}
            keyboardType="numeric"
            maxLength={10}
            onChangeText={[Function]}
            placeholder="Goal"
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
            value={5000}
          />
        }
        cellImageView={null}
        cellStyle="Basic"
        contentContainerStyle={Object {}}
        detail=""
        detailTextStyle={Object {}}
        disableImageResize={false}
        hideSeparator={false}
        highlightActiveOpacity={0.8}
        highlightUnderlayColor="black"
        image={null}
        isDisabled={false}
        leftDetailColor="#007AFF"
        onHighlightRow={null}
        onPress={false}
        onUnHighlightRow={null}
        rightDetailColor="#8E8E93"
        subtitleColor="#000"
        subtitleTextStyle={Object {}}
        title=""
        titleTextColor="#000"
        titleTextStyle={Object {}}
        titleTextStyleDisabled={Object {}}
      />
    </Section>
  </TableView>
</Component>
`;
